InventoryManagementSystem/Repositories/SqlRepository.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System.Data.SqlClient;
+using InventoryManagementSystem.Models;
+
+namespace InventoryManagementSystem.DB
+{
+    internal class SqlRepository : IRepository
+    {
+        private readonly string connectionString;
+
+        public SqlRepository(string connectionString)
+        {
+            this.connectionString = connectionString;
+        }
+
+        public SqlConnection GetConnection()
+        {
+            return new SqlConnection(connectionString);
+        }
+
+        public void AddProduct(Product product)
+        {
+            string insertQuery = @"INSERT INTO Products 
+                                   (
+                                      Name, 
+                                      Price, 
+                                      Quantity
+                                   ) 
+                                   VALUES (
+                                      @Name, 
+                                      @Price, 
+                                      @Quantity
+                                   )";
+
+            using (SqlConnection connection = GetConnection())
```

Comment:
no need to the scope brackets {}, you can use the `using` declaration feature. ex:
`using SqlConnection connection = GetConnection()`